socketioxide/src/errors.rs

```diff
@@ -42,4 +45,56 @@ pub enum AckError {
     /// Internal error
     #[error("internal error: {0}")]
     InternalError(#[from] Error),
+
+    #[error("send channel error: {0:?}")]
+    SendChannel(#[from] SendError),
+}
+
+/// Error type for broadcast operations.
+#[derive(Debug, thiserror::Error)]
+pub enum BroadcastError {
+    /// An error occurred while sending packets.
+    #[error("Sending error: {0:?}")]
+    SendError(Vec<SendError>),
+
+    /// An error occurred while serializing the JSON packet.
+    #[error("Error serializing JSON packet: {0:?}")]
+    Serialize(#[from] serde_json::Error),
+}
+
+impl From<Vec<SendError>> for BroadcastError {
+    /// Converts a vector of `SendError` into a `BroadcastError`.
+    ///
+    /// # Arguments
+    ///
+    /// * `value` - A vector of `SendError` representing the sending errors.
+    ///
+    /// # Returns
+    ///
+    /// A `BroadcastError` containing the sending errors.
+    fn from(value: Vec<SendError>) -> Self {
+        Self::SendError(value)
+    }
+}
+
+/// Error type for sending operations.
+#[derive(thiserror::Error, Debug)]
+pub enum SendError {
+    /// An error occurred while serializing the JSON packet.
+    #[error("Error serializing JSON packet: {0:?}")]
+    Serialize(#[from] serde_json::Error),
+    /// An error occurred during the retry process in the `Retryer`.
+    #[error("Send error: {0:?}")]
+    RetryerError(#[from] RetryerError<SendPacket>),
+}
+
+/// Error type for the `Retryer` struct indicating various failure scenarios during the retry process.
+#[derive(thiserror::Error, Debug)]
+pub enum RetryerError<T: Debug> {
```

Comment:
You don't need any Generic, you can just have `SendPacket`

Reply:
same as the comment above